apps/frontend/src/app/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,161 @@
+import { Fragment, useCallback } from 'react';
+import { Disclosure, Menu, Transition } from '@headlessui/react';
+import { BellIcon, MenuIcon, XIcon } from '@heroicons/react/outline';
+import { useAuth } from '@pasnik/shared/utils-auth';
+import { matchPath } from 'react-router';
+import { NavLink } from 'react-router-dom';
+
+const navigation = [
+  { name: 'Dashboard', href: '' },
+  { name: 'Balances', href: 'balances' },
+  { name: 'Transfers', href: 'transfers' },
+];
+
+function classNames(...classes: string[]) {
+  return classes.filter(Boolean).join(' ');
+}
+
+export function NavBar() {
+  const auth = useAuth();
+  const logoutHandler = useCallback(() => {
+    auth.signOut();
+  }, [auth]);
+  const isCurrent = (href: string) => {
```

Comment:
add useMemo here

Reply:
Are you sure? See usage in the template, I've used useCallback instead